tools/cellprofiler/common.xml

```diff
@@ -88,175 +88,210 @@ import json
 import sys
 import os
 
-
-FOURSPACES=@SPACES@
-NEWLINE=@NEWLINE@
+FOURSPACES = @SPACES@
 
 input_json_path = sys.argv[1]
 
-
 params = json.load(open(input_json_path, "r"))
 
+
 def write_images():
-  filter_images = params['images']['filter_images']
+    filter_images = params['images']['filter_images']
 
-  f.write(NEWLINE)
-  f.write("Images:[module_num:1|svn_version:\\'Unknown\\'|variable_revision_number:2|show_window:False|notes:\\x5B\\'To begin creating your project, use the Images module to compile a list of files and/or folders that you want to analyze. You can also specify a set of rules to include only the desired files in your selected folders.\\'\x5D|batch_state:array(\x5B\x5D, dtype=uint8)|enabled:True|wants_pause:False]"+NEWLINE)
+    _str="\n"
+    _str = "%s%s" %(_str,"Images:[module_num:1|svn_version:\\'Unknown\\'|variable_revision_number:2|show_window:False"
+                         "|notes:\\x5B\\'To begin creating your project, use the Images module to compile a list of files "
+                         "and/or folders that you want to analyze. You can also specify a set of rules to include only "
+                         "the desired files in your selected folders.\\'\x5D|batch_state:array(\x5B\x5D, dtype=uint8)|"
+                         "enabled:True|wants_pause:False]\n")
+    _str = "%s%s%s" %(_str,FOURSPACES,":\n")
+    _str = "%s%s%s%s%s" %(_str,FOURSPACES,"Filter images?:",filter_images,"\n")
+    _str = "%s%s%s" %(_str,FOURSPACES,"Select the rule criteria:and (extension does isimage) (directory doesnot startwith \".\")\n")

```

Comment:
The idea was to save you work :)
Isn't this line, way better? `_str += "    Select the rule criteria:and (extension does isimage) (directory does not start with '.')\n"`

Reply:
that . needs a double quotes otherwise cellprofiler won't work
there're lots of other things similar to this..

if write this way, it is still using + to concatenate.. not string replacement,right?

Reply:
Would this work?
`_str += '    Select the rule criteria:and (extension does isimage) (directory does not start with ".")\n'`

> if write this way, it is still using + to concatenate.. not string replacement, left?

The point was to reduce your operations and clean up the code.